java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogsTest.java

```diff
@@ -0,0 +1,357 @@
+
+        // and everything should work!
+        scaleTranslateButtonOperator.doClick();
+    }
+
+    @Test
+    public void testMoveSelection() {
+
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+
+        layoutEditorDialogs.moveSelection();
+        JFrameOperator jFrameOperator = new JFrameOperator(Bundle.getMessage("TranslateSelection"));
+        Assert.assertNotNull("jFrameOperator 1st exists", jFrameOperator);
+
+        new JButtonOperator(jFrameOperator, Bundle.getMessage("ButtonCancel")).doClick();  // NOI18N
+        //TODO: any way to verify that the dialog is closed?
+
+        // reopen scale track diagram
+        layoutEditorDialogs.moveSelection();
+        jFrameOperator = new JFrameOperator(Bundle.getMessage("TranslateSelection"));
+
+        // get MoveSelection button
+        JButtonOperator moveSelectionButtonOperator = new JButtonOperator(jFrameOperator,
+                Bundle.getMessage("MoveSelection"));
+        Assert.assertNotNull("addNewLabelButtonOperator exists", moveSelectionButtonOperator);
+
+        // try to enter an invalid value in horizontal (x) translation text field
+        JTextFieldOperator horizontalTranslationTextFieldOperator = new JTextFieldOperator(jFrameOperator, 0);
+        Assert.assertNotNull("horizontalTranslationTextFieldOperator exists", horizontalTranslationTextFieldOperator);
+        horizontalTranslationTextFieldOperator.setText("NumberFormatException string");
+
+        Thread misc1 = createModalDialogOperatorThread(Bundle.getMessage("ErrorTitle"),
+                Bundle.getMessage("ButtonOK"), "misc1");  // NOI18N
+        moveSelectionButtonOperator.doClick();
+        JUnitUtil.waitFor(() -> {
+            return !(misc1.isAlive());
+        }, "misc1 finished");
+
+        // now set the horizontal (x) translation text field
+        horizontalTranslationTextFieldOperator.setText("50");
+
+        // try to enter an invalid value in vertical (y) translation text field
+        JTextFieldOperator verticalTranslationTextFieldOperator = new JTextFieldOperator(jFrameOperator, 1);
+        Assert.assertNotNull("verticalTranslationTextFieldOperator exists", verticalTranslationTextFieldOperator);
+        verticalTranslationTextFieldOperator.setText("NumberFormatException string");
+
+        Thread misc2 = createModalDialogOperatorThread(Bundle.getMessage("ErrorTitle"),
+                Bundle.getMessage("ButtonOK"), "misc1");  // NOI18N
+        moveSelectionButtonOperator.doClick();
+        JUnitUtil.waitFor(() -> {
+            return !(misc2.isAlive());
+        }, "misc2 finished");
+
+        // now set the vertical (y) translation text field
+        verticalTranslationTextFieldOperator.setText("100");
+
+        // and everything should work!
+        moveSelectionButtonOperator.doClick();
+    }
+
+    Thread createModalDialogOperatorThread(String dialogTitle, String buttonText, String threadName) {
```

Comment:
This method already exists in a utility class:
https://github.com/JMRI/JMRI/blob/0bf187b6b904f7b43626715756232a3dc9a56bc6/java/test/jmri/util/swing/JemmyUtil.java#L86

Use that class, don’t repeat it here.

Reply:
Excellent, I’ll remove it here then.